### PR TITLE
migrate to vite

### LINKED ISSUE #678: Migrate to VITE
Switch from create-react-app to VITE

### PR DESCRIPTION
**Pull request by IanKrieger**
Resolves #678

### CODE REVIEW

File: src/serviceWorker.ts
Comment (tackley) [inline src/serviceWorker.ts:1]: I'd suggest just deleting this file, we're not calling it. It's a hangover from when create-react-app thought service workers were the future.

File: src/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/// <reference types="vite/client" />
+
+// see https://vitejs.dev/guide/env-and-mode.html#env-files
+
+interface ImportMetaEnv {
+  readonly REACT_APP_SERVER_ADDRESS: string;
+  readonly REACT_APP_BUILD_TIME: string;
+  readonly REACT_APP_GIT_SHA1: string;
+  readonly REACT_APP_QLASS: string;
```
Comment (tackley) [inline src/env.d.ts:9]:
```suggestion
```

Self service doesn't use Qlass (the search estimator).

File: src/user/views/adsManager/views/advanced/components/ads/AdField.tsx [hunk 2]
```diff
@@ -171,7 +170,7 @@ function Preview({ name }: PreviewProp) {
       >
         <Box display="flex" flexDirection="row" justifyContent="center">
           <img
-            src={BraveLogo}
+            src="/brave_logo_icon.png"
```
Comment (tackley) [inline src/user/views/adsManager/views/advanced/components/ads/AdField.tsx:173]: Are you sure this change works?  I don't see that png being processed by vite when I run `npm run build`, which I'd expect it to be. Doing the import of png's is the usual way to to signal to your packager that it needs to be packaged.

**Review by tackley — APPROVED** [review]
This is great!

Two things I'd suggest you check before merging (if you haven't already):

- check that the non-dev version of the site works ok locally (`npm run build` then `npx serve  --single build/`) though that might be tricky with the proxy.
- check that the generated resources are about the same size as they previously were - I'm seeing `build/static/index-0b0093d2.js              1,786.62 kB │ gzip: 547.06 kB │ map: 7,781.55 kB` which is pretty big, but if it's similar to current that's fine.

File: src/user/views/adsManager/views/advanced/components/ads/AdField.tsx (hunk 2 above)
Comment (IanKrieger) [reply src/user/views/adsManager/views/advanced/components/ads/AdField.tsx:173]: I see, it was throwing errors but that was before the addition of the `env.d.ts`. After some experimentation, looks like they are all there

**Comment by IanKrieger** [comment]
@tackley on the build sizes, they are similar with vite taking the win on the `.map`

NEW
```shell
-rw-r--r--   1 macbook  staff   1.7M Apr  3 14:00 index-704043ce.js
-rw-r--r--   1 macbook  staff   7.4M Apr  3 14:00 index-704043ce.js.map
```

OLD
```shell
-rw-r--r--  1 macbook  staff   1.7M Apr  3 14:07 main.bb947443.js
-rw-r--r--  1 macbook  staff   9.2M Apr  3 14:07 main.bb947443.js.map
```

**Pull request merged**